edk2toolext/environment/repo_resolver.py

```diff
@@ -286,6 +286,7 @@ def checkout(abs_file_system_path, dep, repo, update_ok=False, ignore_dep_state_
                 logger.info("We failed to checkout this branch, we'll try to fetch")
                 repo.fetch(branch=branch)
                 repo.checkout(branch=branch)
+            repo.pull()  # Pull to be top of tree
```

Comment:
not sure why we need to pull here.  I worry this could create unintended consequences if repo is dirty.  Maybe we could pull with "--ff-only" but really still not sure why fetch and checkout isn't enough if we are using a fully defined repo path like origin/branch.